### DISCUSSION TITLE
Low m-values for Soundex Comparison Level

### DISCUSSION THREAD

**Discussion by mrazav** [moj-analytical-services/splink#2261] [Q&A]
I am trying to link two tables in link_and_dedupe mode. I have Soundex codes of last names as a separate column in my table. I added a comparison level for exact matches on last name Soundex codes. Although I know that a high percentage of matching pairs should match on last name Soundex, the m-values for this comparison level are low, which is decreasing the total match score.

Here is the comparison levels I'm using for last name:

```
{
"sql_condition": "name_last_l is null or name_last_r is null",

"label_for_charts": "Null",

"is_null_level" : True
},
{
"sql_condition": "name_last_l = name_last_r  or (name_last_l = name_first_r and name_last_r = name_first_l) ",

"label_for_charts": "Exact match and transposition",

},
{
"sql_condition": "jaro_winkler_similarity(name_last_l, name_last_r) >= 0.9",

"label_for_charts": "jaro_winkler_similarity >= 0.9",
},
{
"sql_condition": "name_last_soundex_l = name_last_soundex_r",

"label_for_charts": "Soundex",

"tf_adjustment_weight" : 1.0
},
{
"sql_condition": "jaro_winkler_similarity(name_last_l, name_last_r) >= 0.8",

"label_for_charts": "jaro_winkler_similarity >= 0.8",

},
{
"sql_condition": "else",

"label_for_charts": "all other",

}
```

I have the same situation for another comparison level, `jaro_winkler_similarity >= 0.8`. I expected to have higher m values for this level as well, but this comparison level is leading to a decrease of total match score. 
![image](https://github.com/user-attachments/assets/57ec6a12-0a2a-43a3-b78f-52bf5ad2d064)

Could someone help me understand why the m-values might be low and how I can improve them?
Thank you.

**Comment by lamaeldo** [comment]
Unless I am mistaken, since the m and u values are trained in order, the match weight you are getting for Soundex is not the probability that a soundex match is a true match, it's the possibility that all the soundex match that were not match in the three previous level are a true match. In your case, it seems like it has excluded a lot of the true matches, hence your match weight for soundex is low. Same thing for jaro winkler, it excludes all the previous comparison, including all the jaro winkler > .9
